pkg/authorization/api.go

```diff
@@ -0,0 +1,46 @@
+package authorization
+
+const (
+	// POST method for daemon request authorization
+	AuthZApiRequest = "AuthZPlugin.AuthZReq"
+
+	// POST method for daemon response authorization
+	AuthZApiResponse = "AuthZPlugin.AuthZRes"
+
+	// Name of the interface all AuthZ plugins implement
+	AuthZApiImplements = "authz"
+)
+
+// Request holds data required for authZ plugins
+type Request struct {
+	// User holds the user extracted by AuthN mechanism
+	User string `json:"User,omitempty"`
+
+	// UserAuthNMethod holds the mechanism used to extract user details (e.g., krb)
+	UserAuthNMethod string `json:"UserAuthNMethod,omitempty"`
+
+	// RequestMethod holds the HTTP method (GET/POST/PUT)
+	RequestMethod string `json:"RequestMethod,omitempty"`
+
+	// RequestUri holds the full HTTP uri (e.g., /v1.21/version)
+	RequestUri string `json:"RequestUri,omitempty"`
+
+	// RawRequest stores the raw request sent to the docker daemon
+	RawRequest []byte `json:"RawRequest,omitempty"`
+
+	// RawRequest stores the raw response sent from docker daemon
+	RawResponse []byte `json:"RawResponse,omitempty"`
+}
+
+// Request is the expect object that returns from the authZ plugin
+type Response struct {
```

Comment:
We have decided to go with a simpler approach as the first step and tune / optimize as we go. Current approach is to simply AND between all the plugins and only proceed if the access is approved by all. This can definitely be revisited.